include/gsl/span

```diff
@@ -351,95 +396,100 @@ namespace details
 } // namespace details
 
 // [span], class template span
-template <class ElementType, std::ptrdiff_t Extent>
+template <class ElementType, std::size_t Extent>
 class span
 {
 public:
     // constants and types
     using element_type = ElementType;
     using value_type = std::remove_cv_t<ElementType>;
-    using index_type = std::ptrdiff_t;
+    using size_type = std::size_t;
     using pointer = element_type*;
+    using const_pointer = const element_type*;
     using reference = element_type&;
+    using const_reference = const element_type&;
+    using difference_type = std::ptrdiff_t;
 
-    using iterator = details::span_iterator<span<ElementType, Extent>, false>;
-    using const_iterator = details::span_iterator<span<ElementType, Extent>, true>;
+    using iterator = details::span_iterator<ElementType>;
+    using const_iterator = details::span_iterator<const ElementType>;
     using reverse_iterator = std::reverse_iterator<iterator>;
     using const_reverse_iterator = std::reverse_iterator<const_iterator>;
 
-    using size_type = index_type;
+    using index_type [[deprecated("use size_type instead of index_type")]] = size_type;
 
 #if defined(GSL_USE_STATIC_CONSTEXPR_WORKAROUND)
-    static constexpr const index_type extent{Extent};
+    static constexpr const size_type extent{Extent};
```

Comment:
remove constexpr from the GSL_USE_STATIC_CONSTEXPR_WORKAROUND

Reply:
When GSL_USE_STATIC_CONSTEXPR_WORKAROUND is defined in MSVC constexpr is pushed and redefined as /*constexpr*/, however that is not the case for clang/apple-clang/or g++.  So I'll leave these in.